### PR TITLE
[client] Add client.getDocuments

### PR DESCRIPTION
**Pull request by bjoerge**
This adds a client method for fetching multiple _entire_ documents in one go, using the `/doc` endpoint.

Usage: `client.getDocuments(['foo', 'bar']) => [{_id: 'foo', ...}, {_id: 'bar', ...}]`

Note: This adds a workaround for an issue with the `/doc` HTTP endpoint not yielding `null` values for nonexsting documents in the resulting array. This can lead to nasty bugs, e.g.:

`const [draft, published] = await client.getDocuments([draftId, publishedId])`

Without the workaround, this will assign the published document to the `draft` variable if the draft does not exists.

### CODE REVIEW

File: packages/@sanity/client/README.md [hunk 1]
```diff
@@ -32,14 +32,22 @@ const client = sanityClient({
...
 client.getDocument('bike-123').then(bike => {
   console.log(`${bike.name} (${bike.seats} seats)`)
 })
 ```
 
+### Fetch multiple entire documents in one go
+
+```js
+client.getDocuments(['bike123', 'bike345']).then(([bike123, bike345]) => {
+  console.log(`${bike.name} (${bike.seats} seats)`)
```
Comment (rexxars) [inline packages/@sanity/client/README.md]: The `bike` variable is not defined

File: packages/@sanity/client/README.md [hunk 2]
```diff
@@ -32,14 +32,22 @@ const client = sanityClient({
...
+})
+```
```
Comment (rexxars) [inline packages/@sanity/client/README.md]: We should probably document that missing documents are returned as `null`-values, and that they are ordered in the same order as the input IDs (I'm used to thinking in SQL `SELECT * FROM documents WHERE id IN (a, b, c)`)

File: packages/@sanity/client/test/client.test.js [hunk 3]
```diff
@@ -363,6 +363,43 @@ test('can query for single document', t => {
...
+test('preserves the position of requested documents', t => {
+  nock(projectHost())
+    .get('/v1/data/doc/foo/abc123,abc321')
+    .reply(200, {
+      ms: 123,
+      documents: [{_id: 'abc321', mood: 'tense'}]
+    })
+
+  getClient()
+    .getDocuments(['abc123', 'abc321'])
```
Comment (rexxars) [inline packages/@sanity/client/test/client.test.js]: Perhaps add a third document (which also exist, but is returned in the wrong order), so as to make sure we treat both missing and present documents correctly?

File: packages/@sanity/client/README.md [hunk 4]
```diff
@@ -32,14 +32,22 @@ const client = sanityClient({
 
 Initializes a new Sanity Client. Required options are `projectId` and `dataset`.
 
-### Fetch a single document
+### Fetch an entire single document
```
Comment (rexxars) [inline packages/@sanity/client/README.md]: Why "entire"? What would be the alternative?

File: packages/@sanity/client/README.md (hunk 2 above)
Comment (bjoerge) [reply packages/@sanity/client/README.md]: Good point! Guess I assumed that the order was preserved apart from the omitted null values.

File: packages/@sanity/client/README.md (hunk 4 above)
Comment (bjoerge) [reply packages/@sanity/client/README.md]: The alternative would be a _partial_ document. Maybe "complete", "full" is better? Or just leave it as is?

File: packages/@sanity/client/README.md (hunk 1 above)
Comment (bjoerge) [reply packages/@sanity/client/README.md]: Good catch, thanks!

File: packages/@sanity/client/README.md (hunk 2 above)
Comment (bjoerge) [reply packages/@sanity/client/README.md]: Noted this in the README now, linking to the official doc endpoint docs. Also moved the `getDocument(s)` sections from the top of the README to below the _queries_ section to make it less prominent/significant in the README.

**Pull request merged**